libbeat/processors/cache/mem_store_test.go

```diff
@@ -0,0 +1,378 @@
+		cfg: config{
+			Store: &storeConfig{
+				Memory:   &id{"test"},
+				Capacity: 1000,
+				Effort:   10,
+			},
+			Put: &putConfig{
+				TTL: ptrTo(time.Second),
+			},
+		},
+		want: &memStore{
+			id:     "test",
+			cache:  map[string]*CacheEntry{},
+			refs:   1,
+			ttl:    time.Second,
+			cap:    1000,
+			effort: 10,
+		},
+	},
+	{
+		name: "new_get",
+		cfg: config{
+			Store: &storeConfig{
+				Memory:   &id{"test"},
+				Capacity: 1000,
+				Effort:   10,
+			},
+			Get: &getConfig{},
+		},
+		want: &memStore{
+			id:    "test",
+			cache: map[string]*CacheEntry{},
+			refs:  1,
+			// TTL, capacity and effort are set only by put.
+			ttl:    -1,
+			cap:    -1,
+			effort: -1,
+		},
+	},
+	{
+		name: "new_delete",
+		cfg: config{
+			Store: &storeConfig{
+				Memory:   &id{"test"},
+				Capacity: 1000,
+				Effort:   10,
+			},
+			Delete: &delConfig{},
+		},
+		want: &memStore{
+			id:    "test",
+			cache: map[string]*CacheEntry{},
+			refs:  1,
+			// TTL, capacity and effort are set only by put.
+			ttl:    -1,
+			cap:    -1,
+			effort: -1,
+		},
+	},
+	{
+		name: "new_get_add_put",
+		cfg: config{
+			Store: &storeConfig{
+				Memory:   &id{"test"},
+				Capacity: 1000,
+				Effort:   10,
+			},
+			Get: &getConfig{},
+		},
+		want: &memStore{
+			id:    "test",
+			cache: map[string]*CacheEntry{},
+			// TTL, capacity and effort are set only by put.
+			refs:   1,
+			ttl:    -1,
+			cap:    -1,
+			effort: -1,
+		},
+		steps: []memStoreTestSteps{
+			0: {
+				doTo: func(s *memStore) error {
+					putCfg := config{
+						Store: &storeConfig{
+							Memory:   &id{"test"},
+							Capacity: 1000,
+							Effort:   10,
+						},
+						Put: &putConfig{
+							TTL: ptrTo(time.Second),
+						},
+					}
+					s.add(putCfg)
+					return nil
+				},
+				want: &memStore{
+					id:     "test",
+					cache:  map[string]*CacheEntry{},
+					refs:   2,
+					ttl:    time.Second,
+					cap:    1000,
+					effort: 10,
+				},
+			},
+		},
+	},
+	{
+		name: "ensemble",
+		cfg: config{
+			Store: &storeConfig{
+				Memory:   &id{"test"},
+				Capacity: 1000,
+				Effort:   10,
+			},
+			Get: &getConfig{},
+		},
+		want: &memStore{
+			id:    "test",
+			cache: map[string]*CacheEntry{},
+			refs:  1,
+			// TTL, capacity and effort are set only by put.
+			ttl:    -1,
+			cap:    -1,
+			effort: -1,
+		},
+		steps: []memStoreTestSteps{
+			0: {
+				doTo: func(s *memStore) error {
+					putCfg := config{
+						Store: &storeConfig{
+							Memory:   &id{"test"},
+							Capacity: 1000,
+							Effort:   10,
+						},
+						Put: &putConfig{
+							TTL: ptrTo(time.Second),
+						},
+					}
+					s.add(putCfg)
+					return nil
+				},
+				want: &memStore{
+					id:     "test",
+					cache:  map[string]*CacheEntry{},
+					refs:   2,
+					ttl:    time.Second,
+					cap:    1000,
+					effort: 10,
+				},
+			},
+			1: {
+				doTo: func(s *memStore) error {
+					s.Put("one", 1)
+					s.Put("two", 2)
+					s.Put("three", 3)
```

Comment:
```suggestion
					_ = s.Put("three", 3)
```

Reply:
This [should not](https://github.com/elastic/beats/blob/ea3b9e34bdea09bba0f951a7c4cb562bb037b09a/.golangci.yml#L81) be flagged.